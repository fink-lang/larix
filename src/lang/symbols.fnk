{symbol: parser_symbol} = import '@fink/prattler/symbols'
{next_loc} = import '@fink/prattler'
{indentation} = import './indentation'


symbol = fn id:
  {
    ...parser_symbol(id),

    lbp: fn lbp: fn ctx:
      {start: {column}} = next_loc(ctx)

      match true:
        column <= indentation(ctx): 0
        else: lbp
  }
