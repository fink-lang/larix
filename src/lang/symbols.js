import {symbol as parser_symbol} from '@fink/prattler/symbols';
import {next_loc} from '@fink/prattler';
import {indentation} from './indentation';


export const symbol = (id)=> ({
  ...parser_symbol(id),

  lbp: (lbp)=> (ctx)=> {
    if (next_loc(ctx).start.column <= indentation(ctx)) {
      return 0;
    }
    return lbp;
  }
});
