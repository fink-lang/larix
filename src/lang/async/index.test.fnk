{describe, it, eq, snapshot} = import '../../testing/jest'

{parse_expr} = import '../../'


describe:: 'async', fn:

  it:: 'parses async: await foobar', fn:

    parse_expr(`await foobar`) eq snapshot

