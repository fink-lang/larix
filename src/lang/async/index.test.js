import {other_token} from '@fink/prattler/symbols';

import {parse_expr} from '../../';


describe('assignment', ()=> {

  it('parses single line: foo = bar', ()=> {
    expect(
      parse_expr(`await foobar`)
    ).toMatchSnapshot();
  });
});
