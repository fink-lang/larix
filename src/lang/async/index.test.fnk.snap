// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async parses async: await foobar 1`] = `
await (1:0-1:12)
  ident (1:6-1:12) foobar
`;
